Test case failure

=================================== FAILURES ===================================
_______________ TestCommandExecutorAsync.test_git_branch_command _______________

self = <test_command_executor_async.TestCommandExecutorAsync object at 0x7f2babf0f190>
executor = <mcp_tools.command_executor.executor.CommandExecutor object at 0x7f2babf7a0d0>

    async def test_git_branch_command(self, executor):
        """Test executing a git command that might be used frequently"""
        # This depends on the test running in a git repo directory
        if not os.path.exists(".git") and not os.path.exists("../.git"):
            pytest.skip("Not in a git repository")
    
        # Store original directory
        original_dir = os.getcwd()
        try:
            # Change to the directory containing this test file
            test_file_dir = os.path.dirname(os.path.abspath(__file__))
            os.chdir(test_file_dir)
    
            # Find the git root directory (assuming we're in a git repo)
            response = await executor.execute_async("git rev-parse --show-toplevel")
            token = response["token"]
            result = await executor.wait_for_process(token, timeout=5.0)
    
            assert result["status"] == "completed"
            assert result["output"].strip() != "", "Not in a git repository"
            git_root = result["output"].strip()
    
            # Change to git root directory
            os.chdir(git_root)
    
            # Run the branch command
            cmd = "git branch --show-current"
            response = await executor.execute_async(cmd)
            token = response["token"]
    
            # Wait for completion
            result = await executor.wait_for_process(token, timeout=5.0)
    
            # Verify it completed
            assert result["status"] == "completed"
>           assert result["output"].strip() != "", "Branch name should not be empty"
E           AssertionError: Branch name should not be empty
E           assert '' != ''
E            +  where '' = <built-in method strip of str object at 0x7f2bb093f0c0>()
E            +    where <built-in method strip of str object at 0x7f2bb093f0c0> = ''.strip

mcp_tools/tests/test_command_executor_async.py:430: AssertionError
------------------------------ Captured log setup ------------------------------
INFO     mcp_tools.command_executor.executor:executor.py:43 Initialized CommandExecutor | Context: {"temp_dir": "/tmp", "os_type": "linux", "timestamp": "2025-06-05T07:10:21.309875+00:00"}
DEBUG    asyncio:selector_events.py:54 Using selector: EpollSelector
------------------------------ Captured log call -------------------------------
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Created temp files | Context: {"stdout_path": "/tmp/cmd_out_awkv0hwu", "stderr_path": "/tmp/cmd_err_ed3q096t", "timestamp": "2025-06-05T07:10:21.3[105](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:106)92+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Starting async command | Context: {"command": "git rev-parse --show-toplevel", "token": "9192d870-aaa3-41bb-bd3f-1d792d0640f8", "timestamp": "2025-06-05T07:10:21.3[106](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:107)82+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Started async command | Context: {"command": "git rev-parse --show-toplevel", "token": "9192d870-aaa3-41bb-bd3f-1d792d0640f8", "pid": 2755, "timestamp": "2025-06-05T07:10:21.311311+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Waiting for process to complete | Context: {"token": "9192d870-aaa3-41bb-bd3f-1d792d0640f8", "pid": 2755, "timeout": 5.0, "timestamp": "2025-06-05T07:10:21.3[114](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:115)22+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Process completed after waiting | Context: {"token": "9192d870-aaa3-41bb-bd3f-1d792d0640f8", "pid": 2755, "returncode": 0, "timestamp": "2025-06-05T07:10:21.313414+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Read temp file: /tmp/cmd_out_awkv0hwu | Context: {"path": "/tmp/cmd_out_awkv0hwu", "content_length": 26, "content_sample": "/home/runner/work/mcp/mcp\n", "timestamp": "2025-06-05T07:10:21.313530+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Read temp file: /tmp/cmd_err_ed3q096t | Context: {"path": "/tmp/cmd_err_ed3q096t", "content_length": 0, "content_sample": "", "timestamp": "2025-06-05T07:10:21.313624+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Cleaning up temp files for PID 2755 | Context: {"pid": 2755, "stdout_path": "/tmp/cmd_out_awkv0hwu", "stderr_path": "/tmp/cmd_err_ed3q096t", "from_monitor": false, "timestamp": "2025-06-05T07:10:21.313689+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Removed stdout temp file: /tmp/cmd_out_awkv0hwu | Context: {"pid": 2755, "path": "/tmp/cmd_out_awkv0hwu", "timestamp": "2025-06-05T07:10:21.313779+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Removed stderr temp file: /tmp/cmd_err_ed3q096t | Context: {"pid": 2755, "path": "/tmp/cmd_err_ed3q096t", "timestamp": "2025-06-05T07:10:21.313861+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Returning final result | Context: {"token": "9192d870-aaa3-41bb-bd3f-1d792d0640f8", "pid": 2755, "returncode": 0, "stdout_length": 26, "stderr_length": 0, "timestamp": "2025-06-05T07:10:21.313920+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Created temp files | Context: {"stdout_path": "/tmp/cmd_out_e8mglc_6", "stderr_path": "/tmp/cmd_err_t0dlzqeh", "timestamp": "2025-06-05T07:10:21.314069+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Starting async command | Context: {"command": "git branch --show-current", "token": "f2e9b928-954a-4[135](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:136)-ac05-67ac578aeda9", "timestamp": "2025-06-05T07:10:21.314139+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Started async command | Context: {"command": "git branch --show-current", "token": "f2e9b928-954a-4135-ac05-67ac578aeda9", "pid": 2758, "timestamp": "2025-06-05T07:10:21.314571+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Waiting for process to complete | Context: {"token": "f2e9b928-954a-4135-ac05-67ac578aeda9", "pid": 2758, "timeout": 5.0, "timestamp": "2025-06-05T07:10:21.3[146](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:147)53+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Process completed after waiting | Context: {"token": "f2e9b928-954a-4135-ac05-67ac578aeda9", "pid": 2758, "returncode": 0, "timestamp": "2025-06-05T07:10:21.316392+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Read temp file: /tmp/cmd_out_e8mglc_6 | Context: {"path": "/tmp/cmd_out_e8mglc_6", "content_length": 0, "content_sample": "", "timestamp": "2025-06-05T07:10:21.316503+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Read temp file: /tmp/cmd_err_t0dlzqeh | Context: {"path": "/tmp/cmd_err_t0dlzqeh", "content_length": 0, "content_sample": "", "timestamp": "2025-06-05T07:10:21.316593+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Cleaning up temp files for PID 2758 | Context: {"pid": 2758, "stdout_path": "/tmp/cmd_out_e8mglc_6", "stderr_path": "/tmp/cmd_err_t0dlzqeh", "from_monitor": false, "timestamp": "2025-06-05T07:10:21.316656+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Removed stdout temp file: /tmp/cmd_out_e8mglc_6 | Context: {"pid": 2758, "path": "/tmp/cmd_out_e8mglc_6", "timestamp": "2025-06-05T07:10:21.316731+00:00"}
DEBUG    mcp_tools.command_executor.executor:executor.py:43 Removed stderr temp file: /tmp/cmd_err_t0dlzqeh | Context: {"pid": 2758, "path": "/tmp/cmd_err_t0dlzqeh", "timestamp": "2025-06-05T07:10:21.316809+00:00"}
INFO     mcp_tools.command_executor.executor:executor.py:43 Wait process: Returning final result | Context: {"token": "f2e9b928-954a-4135-ac05-67ac578aeda9", "pid": 2758, "returncode": 0, "stdout_length": 0, "stderr_length": 0, "timestamp": "2025-06-05T07:10:21.316865+00:00"}
---------------------------- Captured log teardown -----------------------------
DEBUG    asyncio:selector_events.py:54 Using selector: EpollSelector
=============================== warnings summary ===============================
mcp_tools/tests/test_tasks.py::test_tasks
  /home/runner/work/mcp/mcp/.venv/lib/python3.11/site-packages/_pytest/python.py:[148](https://github.com/plusplusoneplusplus/mcp/actions/runs/15460723363/job/43521317744#step:8:149): PytestUnhandledCoroutineWarning: async def functions are not natively supported and have been skipped.
  You need to install a suitable plugin for your async framework, for example:
    - anyio
    - pytest-asyncio
    - pytest-tornasync
    - pytest-trio
    - pytest-twisted
    warnings.warn(PytestUnhandledCoroutineWarning(msg.format(nodeid)))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED mcp_tools/tests/test_command_executor_async.py::TestCommandExecutorAsync::test_git_branch_command - AssertionError: Branch name should not be empty
assert '' != ''
 +  where '' = <built-in method strip of str object at 0x7f2bb093f0c0>()
 +    where <built-in method strip of str object at 0x7f2bb093f0c0> = ''.strip
============= 1 failed, 87 passed, 9 skipped, 1 warning in 39.63s ==============
✗ Some tests for mcp_tools failed: 1 failed, 87 passed, 9 skipped

--- a/mcp_tools/tests/test_command_executor_async.py
+++ b/mcp_tools/tests/test_command_executor_async.py
@@ -427,11 +427,24 @@ class TestCommandExecutorAsync:
 
             # Verify it completed
             assert result["status"] == "completed"
-            assert result["output"].strip() != "", "Branch name should not be empty"
-            # Branch name should be a valid git branch name (no spaces or special chars except - _ /)
-            assert all(
-                c.isalnum() or c in "-_/" for c in result["output"].strip()
-            ), "Invalid branch name characters"
+
+            # Check if we're in detached HEAD state (common in CI environments)
+            branch_name = result["output"].strip()
+            if branch_name == "":
+                # In detached HEAD state, verify we can still get commit info
+                response = await executor.execute_async("git rev-parse HEAD")
+                token = response["token"]
+                commit_result = await executor.wait_for_process(token, timeout=5.0)
+
+                assert commit_result["status"] == "completed"
+                commit_hash = commit_result["output"].strip()
+                assert len(commit_hash) == 40, "Should get a valid commit hash"
+                assert all(c in "0123456789abcdef" for c in commit_hash), "Invalid commit hash characters"
+            else:
+                # On a named branch, verify branch name is valid
+                assert all(
+                    c.isalnum() or c in "-_/" for c in branch_name
+                ), "Invalid branch name characters"
         finally:
             # Restore original directory
             os.chdir(original_dir)

